jobs/keepalived/templates/bin/action.sh.erb

```diff
@@ -32,5 +37,24 @@ if [[ "$iaas_name" == "aws" ]]; then
     sudo ip addr del $virtual_ip/32 dev eth0
     set -e
   fi
+elif [[ "$iaas_name" == "openstack"  ]]; then
+  # shellcheck disable=SC1091
+  source /var/vcap/jobs/keepalived/bin/utils/openstack/generate_token
+  source /var/vcap/jobs/keepalived/bin/utils/openstack/port_operations
+  this_port_id=$(get_port_id "${SOURCE_IP}")
+  if [[ "$STATE" == "MASTER" ]]; then
+    port_update_response=$(update_port "${this_port_id}")
+    peer_port_id=$(get_port_id "${PEER_IP}")
+    port_update_response=$(update_port "${peer_port_id}")
+    port_update_response=$(update_port "${this_port_id}" "${virtual_ip}")
```

Comment:
Why is update_port method called multiple times with different parameters?
Didnt understand this part of code

Reply:
Detaching the virtual ip from both master and slave's ports and updating the promoted master's port with virtual ip. Reason is 1. clearing allowed address pair(AAP) from slave (obvious) 2. clearing master's one also , as two neutron ports can be updated with same virtual ip as AAP. But considered the latest one. In case of fail-over promoted master's port would be stale if not updated.
update_port "${peer_port_id}" ==> Detaches the allowed address pair
update_port "${this_port_id}" "${virtual_ip}" ==> Attaches AAP with virtual ip